### PR TITLE
Created rent model + added foreign key to user + book

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary
 [Crear modelo de Rent](https://trello.com/c/6fnWL8lV/19-crear-modelo-de-rent)

A model for Rent was implemented with references to Book and User. In the User model, a reference for its rents was added as well.

## Notes
Other than the migration to create the rent table, another migration was needed to be able to reference a User's rents. Not null constraints were also added to the validated columns.

### Creating a rent
rent-created-rails-c

### Obtaining a User's rents
rent-user-association

### CODE REVIEW

File: app/models/rent.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Rent < ApplicationRecord
+  belongs_to :user
+  belongs_to :book
```
Comment (marcegl91) [inline app/models/rent.rb]: Book doesn't have the has_one/has_many relationship that uses this belongs_to :'(, so I won't be able to ask to a book which rents are associated.

File: db/migrate/20190724174249_add_rents_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddRentsToUser < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :users, :rents, foreign_key: true
```
Comment (marcegl91) [inline db/migrate/20190724174249_add_rents_to_user.rb]: Could you explain what this migration is doing and why is needed?

Comment (tagonzalez) [reply db/migrate/20190724174249_add_rents_to_user.rb]: Of course!
This migration adds a rents attribute to the User table as so to reference all the rents that a User has made. They should appear as references by id. It also specifies that that column will contain foreign key.

It is needed to be able to ask for all the rents from the User. If not, one could ask the Rents controller to retrieve the rents of a particular User, but I think that that may perform less efficiently than asking from a User if there is no index for user in the Rents table.

When one asks from the User controller, the DB will first query for the User id and then with these references already in the table find them by id (which by default has an index) in the Rents table.
When one asks from Rents, the DB will have to filter all the Rents records (which from context I believe that can be a very large number) for Rents belonging to a User id.

File: app/models/rent.rb (hunk 1 above)
Comment (tagonzalez) [reply app/models/rent.rb]: From reading the Trello card it was explicit that the User should have a this relationship since it is important to be able to ask the rents from a User. Since there was no mention of the same functionality for a book, I thought I should skip it.

However now I realize by reading the [Apiary](https://wbooksapi.docs.apiary.io/#) that this is necessary.
The relationship and foreign keys were added in latest commit.

File: db/migrate/20190724174249_add_rents_to_user.rb (hunk 2 above)
Comment (tagonzalez) [reply db/migrate/20190724174249_add_rents_to_user.rb]: After our conversation on slack, a new migration was added to remove these foreign keys  and is available on the latest commits.

File: db/migrate/20190726181130_remove_rents_from_users_and_books.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class RemoveRentsFromUsersAndBooks < ActiveRecord::Migration[5.1]
+  def change
+    remove_foreign_key :users, :rents
```
Comment (marcegl91) [inline db/migrate/20190726181130_remove_rents_from_users_and_books.rb]: I think it was better to use remove_reference, but I'm trusting your schema.rb and I don't see anything wrong so I think this is ok.

More info: https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-remove_reference

Also, I think that deleting/modifying the migration file was enough, because you were working in your local environment and you can rollback migrations from the database (ie, when you say to Rails i want to rollback to some migration, if the migration was done with a change method, Rails automatically knows how to undo that migration):

https://guides.rubyonrails.org/active_record_migrations.html#rolling-back
https://guides.rubyonrails.org/active_record_migrations.html#using-the-change-method

If you were to push to a branch where other people will do a pull and then have to run your migration, you'd have to write a new migration undoing your changes so everyone is in sync.

tl;dr: it's ok :) .
